wallet-new/test/unit/Test/Spec/Addresses.hs

```diff
@@ -123,6 +126,38 @@ prepareAddressFixture n = do
                 let SliceOf{..} = Addresses.getAddresses (RequestParams pp) db'
                 return . map AddressFixture $ paginatedSlice
 
+prepareAddressesFixture
+    :: Int  -- ^ Number of Accounts to create.
+    -> Int  -- ^ Number of 'Address per account to create.
+    -> Fixture.GenPassiveWalletFixture (M.Map V1.AccountIndex [V1.WalletAddress])
+prepareAddressesFixture acn adn = do
+    spendingPassword <- Fixture.genSpendingPassword
+    newWalletRq <- WalletLayer.CreateWallet <$> Wallets.genNewWalletRq spendingPassword
+    return $ \pw -> do
+        let newAcc (n :: Int) = (V1.NewAccount spendingPassword ("My Account " <> show n))
+        Right v1Wallet <- Wallets.createWallet pw newWalletRq
+        forM_ [1..acn] $ \n ->
+            Accounts.createAccount pw (V1.walId v1Wallet) (newAcc n)
+        -- Get all the available accounts
+        db <- Kernel.getWalletSnapshot pw
+        let Right accs = Accounts.getAccounts (V1.walId v1Wallet) db
+        let accounts = IxSet.toList accs
+        length accounts `shouldBe` (acn + 1)
```

Comment:
I would recommend flipping the logic and doing something like:

```
prepareAddressFixture acn adn = 
    -- We recompute the values to account for the fact that creating a new wallet
    -- @always@ come with a fresh account, and this account always have a fresh address
    -- associated with it.
    let requestedAccounts = max 0 (acn - 1)
    let requestedAddresses = max 0 (adn - 1)
```

Then, you can use those two let bindings all over the place without having to worry about +1/-1 arithmetic, both in the fixtures and in the tests. Furthermore, you segregate the comment on _why_ we do this exactly in one point in the code, without this assumption being scattered all over the place.

Reply:
I'm not sure I understand what are the semantics of `requestedAddresses`. Some accounts will have requestedAddresses addresses, while the first account will have requestedAddresses+1. Also requestedAccounts should never be 0, as one account is created by default.